lib/services/history/sources/remote_history.ts

```diff
@@ -0,0 +1,78 @@
+import { SNApiService } from "@Lib/services/api/api_service";
+import { SNProtocolService } from "@Lib/services/protocol_service";
+import { ItemHistory } from "../item_history";
+import { HttpResponse } from "@Lib/services/api/http_service";
+import { RawPayload, CreateMaxPayloadFromAnyObject, CreateSourcedPayloadFromObject } from "@Lib/protocol/payloads/generator";
+import { ItemHistorySource, ItemHistoryEntry } from "../item_history_entry";
+import { PayloadSource } from "@Lib/protocol/payloads";
+
+type RevisionResponse = {
+  error?: any
+  revision?: RawPayload
```

Comment:
Revisions are not exactly RawPayloads. They share 99% of the same properties, but they still differ in that a revision has an `item_uuid` property. You should instead create a new RawRevision type that extends RawPayload.